# HG changeset patch
# User Rob Mahurin  <rob@utk.edu>
# Date 1238048965 -3600
#      Thu Mar 26 07:29:25 2009 +0100
# Branch release-3-0-x
# Node ID af4fa72ee2504ab057a215aadc4eb252d8a63480
# Parent  0eaaf7351ef55d613b76bb3371b7d2ebc9671e71
ensure row vector from create_set

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-03-25  Rob Mahurin  <rob@utk.edu>
+
+       * set/create_set.m: return a row vector.  From M. Caliari.
+
 2009-03-25  Jaroslav Hajek  <highegg@gmail.com>
 
 	* plot/clf.m: Fix due to M. Caliari.
 
 2009-02-24  Rob Mahurin  <rob@utk.edu>	
 
 	* plot/legend.m: Fix legend order.  Add explicit colors to demos.
 	From Marco Caliari <marco.caliari@univr.it>
diff --git a/scripts/set/create_set.m b/scripts/set/create_set.m
--- a/scripts/set/create_set.m
+++ b/scripts/set/create_set.m
@@ -47,16 +47,17 @@
 
 function y = create_set (x, rows_opt)
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
   endif
   
   if (nargin == 1)
-    y = unique (x).';
+    y = unique (x);
+    y = y(:).';
   elseif (strcmpi (rows_opt, "rows"))
     y = unique (x, "rows");
   else
     error ("create_set: expecting \"rows\" as second argument");
   endif
 
 endfunction
